### PR TITLE
feat: update txt2img L2 api

### PR DESCRIPTION
**Pull request by NingLu**
## Description

Update txt2img L2 api according to recent changes

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ - ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ - ] This change requires a documentation update

### CODE REVIEW

File: middleware_api/lambda/inference_v2/inference_api.py [hunk 1]
```diff
@@ -191,6 +191,66 @@ def run_inference(event, _):
...
+    prepare_resp = prepare_inference(prepare_infer_event, context)
+
+    if 'inference' not in prepare_resp or 'api_params_s3_location' not in prepare_resp['inference']:
+        return {
+            'status': 500,
+            'err': f'fail to prepare inference for {request_id}, no s3 location is generated'
+        }
+
+    s3_location = prepare_resp['inference']['api_params_s3_location']
+    bucket_and_key = s3_location.replace("s3://", "").split("/", 1)
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference_v2/inference_api.py:222]: we've a function to get bucket name and key from a s3 url. model_and_train.common_tools.py#L85:split_s3_path, let's move it to common.util and reuse the method

File: middleware_api/lambda/common/template/inferenceTemplate.json [hunk 2]
```diff
@@ -1,148 +1,76 @@
...
+    "script_name": null,
+    "send_images": true,
+    "save_images": false,
+    "alwayson_scripts": {
+        "controlnet": {
+            "args": [
+                {
+                    "enabled": false,
+                    "module": "none",
+                    "model": "None",
```
Comment (alvindaiyan) [inline middleware_api/lambda/common/template/inferenceTemplate.json:52]: difference "none" vs "None" vs null?

File: middleware_api/lambda/inference_v2/inference_api.py [hunk 3]
```diff
@@ -191,6 +191,66 @@ def run_inference(event, _):
     }
 
 
+# POST /inference-api
+def inference_l2(raw_event, context):
+    request_id = context.aws_request_id
+    if 'task_type' not in raw_event or 'sagemaker_endpoint_name' not in raw_event or 'models' not in raw_event:
+        return {
+            'status': 400,
+            'err': f'task_type, sagemaker_endpoint_name, models should be in the request body'
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference_v2/inference_api.py:200]: sagemaker_endpoint_name should be {sagemaker_endpoint_name}?

File: middleware_api/lambda/inference_v2/inference_api.py [hunk 4]
```diff
@@ -191,6 +191,66 @@ def run_inference(event, _):
     }
 
 
+# POST /inference-api
+def inference_l2(raw_event, context):
```
Comment (alvindaiyan) [inline middleware_api/lambda/inference_v2/inference_api.py:195]: shall we have a class defined for raw_event?

**Review by alvindaiyan — CHANGES_REQUESTED** [review]

File: middleware_api/lambda/common/template/inferenceTemplate.json (hunk 2 above)
Comment (NingLu) [reply middleware_api/lambda/common/template/inferenceTemplate.json:52]: This is generated from webui, i have tested it on my local env, it works fine

File: middleware_api/lambda/inference_v2/inference_api.py (hunk 1 above)
Comment (NingLu) [reply middleware_api/lambda/inference_v2/inference_api.py]: Done, use the function in util

File: middleware_api/lambda/inference_v2/inference_api.py (hunk 4 above)
Comment (NingLu) [reply middleware_api/lambda/inference_v2/inference_api.py:195]: The api schema is inherited from webui fastapi with default value, it contains around 40 parameters, we can add them in further pr

**Review by alvindaiyan — APPROVED** [review]

**Pull request merged**